### PR TITLE
python3Packages.pyjwt: 2.0.1 -> 2.1.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 2.1.0

Change log: https://github.com/jpadilla/pyjwt/blob/master/CHANGELOG.rst#v210

- With 2.0.0 support for Python 2 [was dropped](https://github.com/jpadilla/pyjwt/blob/master/CHANGELOG.rst#v200).
- Switch to `pytestCheckHook`
- Run all tests

Related to #121883

ZHF: #122042

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121891` at f0cc3b12 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>24 packages marked as broken and skipped:</summary>
  <ul>
    <li>bareos</li>
    <li>python38Packages.arviz</li>
    <li>python38Packages.dm-sonnet</li>
    <li>python38Packages.graph_nets</li>
    <li>python38Packages.ignite</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.pymc3</li>
    <li>python38Packages.python-smarttub</li>
    <li>python38Packages.pytorchWithCuda</li>
    <li>python38Packages.rl-coach</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>22 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/44343c4e703b01205e54dfd5d67c5fd5">python38Packages.atlassian-python-api</a></li>
    <li><a href="https://gist.github.com/40a3353e69c1b6c449d4f34f6aad0e90">python38Packages.djangorestframework-jwt</a></li>
    <li><a href="https://gist.github.com/cdd609f760f18f38cb3207b9bf156f4d">python38Packages.exchangelib</a></li>
    <li><a href="https://gist.github.com/7e3e42dd600498bde6dba610082c6865">python38Packages.flask-appbuilder</a></li>
    <li><a href="https://gist.github.com/c76d4f0debc774d5bd09731b62cdcee9">python38Packages.globus-sdk</a></li>
    <li><a href="https://gist.github.com/00a4bdb7c4bf28d9583dc2e69a0503b4">python38Packages.mwoauth</a></li>
    <li><a href="https://gist.github.com/e2b45eefa8ecbdc96faa1989ab4d98ab">python38Packages.oauthenticator</a></li>
    <li><a href="https://gist.github.com/35f8a4d3aa3dc14a2856cbdf978b2281">python38Packages.pyflume</a></li>
    <li><a href="https://gist.github.com/a2cf807b440e3c1c2c9acb5528b38e12">python38Packages.snowflake-connector-python</a></li>
    <li><a href="https://gist.github.com/f025aef292760b992e460377f1098a29">python38Packages.snowflake-sqlalchemy</a></li>
    <li><a href="https://gist.github.com/e4f0078488ae8e911544c65f9aa25610">python38Packages.twilio</a></li>
    <li><a href="https://gist.github.com/21d3fe5e80e1166918a0f437fd9bffd6">python39Packages.atlassian-python-api</a></li>
    <li><a href="https://gist.github.com/1a8c3a24dc9f3c7e66f27583d67b2bb0">python39Packages.djangorestframework-jwt</a></li>
    <li><a href="https://gist.github.com/262aec0a24dc5590b3f08f65aa66c77a">python39Packages.exchangelib</a></li>
    <li><a href="https://gist.github.com/ec87beefc169995971ab8d1a373452f6">python39Packages.flask-appbuilder</a></li>
    <li><a href="https://gist.github.com/c0d6e3fba6a8379608661bc8ebbb276d">python39Packages.globus-sdk</a></li>
    <li><a href="https://gist.github.com/7372acf063fd1d0a469e8aaec854e852">python39Packages.mwoauth</a></li>
    <li><a href="https://gist.github.com/cab1f45d687ffe5bec0c88c1645aa459">python39Packages.oauthenticator</a></li>
    <li><a href="https://gist.github.com/8420a5a98c075883d4e2873f1d4709e5">python39Packages.pyflume</a></li>
    <li><a href="https://gist.github.com/721bf628fd5f0497b182a90d0d598255">python39Packages.snowflake-connector-python</a></li>
    <li><a href="https://gist.github.com/088c05a9cefb75bd928d598d2ce7cb85">python39Packages.snowflake-sqlalchemy</a></li>
    <li><a href="https://gist.github.com/0f624fe30fe1ae0ccd317de25e3d95a8">python39Packages.twilio</a></li>
  </ul>
</details>
<details>
  <summary>337 packages skipped due to time constraints:</summary>
  <ul>
    <li>apache-airflow</li>
    <li>appdaemon</li>
    <li>apprise (python38Packages.apprise)</li>
    <li>azure-cli</li>
    <li>beets</li>
    <li>beetsExternalPlugins.alternatives</li>
    <li>beetsExternalPlugins.copyartifacts</li>
    <li>beetsExternalPlugins.extrafiles</li>
    <li>breezy (python38Packages.breezy)</li>
    <li>cabal2nix</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>75 packages built successfully:</summary>
  <ul>
    <li>alerta-server</li>
    <li>buildbot (python38Packages.buildbot)</li>
    <li>buildbot-full (python38Packages.buildbot-full)</li>
    <li>buildbot-ui (python38Packages.buildbot-ui)</li>
    <li>cligh</li>
    <li>cloud-init</li>
    <li>eolie</li>
    <li>mathlibtools (python38Packages.mathlibtools)</li>
    <li>matrix-synapse</li>
    <li>pleroma-bot</li>
    <li>python38Packages.PyGithub</li>
    <li>python38Packages.adal</li>
    <li>python38Packages.azure-core</li>
    <li>python38Packages.azure-datalake-store</li>
    <li>python38Packages.azure-identity</li>
    <li>python38Packages.azure-storage-blob</li>
    <li>python38Packages.criticality-score</li>
    <li>python38Packages.discogs_client</li>
    <li>python38Packages.django-oauth-toolkit</li>
    <li>python38Packages.djangorestframework-simplejwt</li>
    <li>python38Packages.flask-jwt-extended</li>
    <li>python38Packages.gidgethub</li>
    <li>python38Packages.google-auth-oauthlib</li>
    <li>python38Packages.homepluscontrol</li>
    <li>python38Packages.jupyterhub</li>
    <li>python38Packages.keep</li>
    <li>python38Packages.kubernetes</li>
    <li>python38Packages.lazr-restfulclient</li>
    <li>python38Packages.msal</li>
    <li>python38Packages.msal-extensions</li>
    <li>python38Packages.msrest</li>
    <li>python38Packages.msrestazure</li>
    <li>python38Packages.oauthlib</li>
    <li>python38Packages.ondilo</li>
    <li>python38Packages.pyatmo</li>
    <li>python38Packages.pyfxa</li>
    <li>python38Packages.pyjwt</li>
    <li>python38Packages.requests_oauthlib</li>
    <li>python38Packages.ring-doorbell</li>
    <li>python38Packages.webexteamssdk</li>
    <li>python39Packages.PyGithub</li>
    <li>python39Packages.adal</li>
    <li>python39Packages.azure-core</li>
    <li>python39Packages.azure-datalake-store</li>
    <li>python39Packages.azure-identity</li>
    <li>python39Packages.azure-storage-blob</li>
    <li>python39Packages.buildbot</li>
    <li>python39Packages.buildbot-full</li>
    <li>python39Packages.buildbot-ui</li>
    <li>python39Packages.criticality-score</li>
    <li>python39Packages.discogs_client</li>
    <li>python39Packages.django-oauth-toolkit</li>
    <li>python39Packages.djangorestframework-simplejwt</li>
    <li>python39Packages.flask-jwt-extended</li>
    <li>python39Packages.gidgethub</li>
    <li>python39Packages.google-auth-oauthlib</li>
    <li>python39Packages.homepluscontrol</li>
    <li>python39Packages.jupyterhub</li>
    <li>python39Packages.keep</li>
    <li>python39Packages.kubernetes</li>
    <li>python39Packages.lazr-restfulclient</li>
    <li>python39Packages.mathlibtools</li>
    <li>python39Packages.msal</li>
    <li>python39Packages.msal-extensions</li>
    <li>python39Packages.msrest</li>
    <li>python39Packages.msrestazure</li>
    <li>python39Packages.oauthlib</li>
    <li>python39Packages.ondilo</li>
    <li>python39Packages.pyatmo</li>
    <li>python39Packages.pyfxa</li>
    <li>python39Packages.pyjwt</li>
    <li>python39Packages.requests_oauthlib</li>
    <li>python39Packages.ring-doorbell</li>
    <li>python39Packages.webexteamssdk</li>
    <li>sourcehut.dispatchsrht</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/msal/default.nix:29:0:
```
   |
29 |     description = "The Microsoft Authentication Library (MSAL) for Python library enables your app to access the Microsoft Cloud by supporting authentication of users with Microsoft Azure Active Directory accounts (AAD) and Microsoft Accounts (MSA) using industry standard OAuth2 and OpenID Connect";
   | ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Review by dotlambda — APPROVED** [review]

File: pkgs/development/python-modules/snowflake-connector-python/default.nix [hunk 1]
```diff
@@ -10,7 +11,6 @@
 , idna
 , ijson
 , isPy3k
```
Comment (rmcgibbo) [inline pkgs/development/python-modules/snowflake-connector-python/default.nix:13]: This argument (`isPy3k`) is unused.

**Review by rmcgibbo — APPROVED** [review]
Lots of fixes here. LGTM.

----------------

Result of `nixpkgs-review pr 121891` at 2162e06a run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>20 packages marked as broken and skipped:</summary>
  <ul>
    <li>bareos</li>
    <li>python38Packages.arviz</li>
    <li>python38Packages.dm-sonnet</li>
    <li>python38Packages.graph_nets</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.pymc3</li>
    <li>python38Packages.python-smarttub</li>
    <li>python38Packages.rl-coach</li>
    <li>python38Packages.shap</li>
    <li>python38Packages.skorch</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>11 packages failed to build:</summary>
  <ul>
    <li>python38Packages.atlassian-python-api</li>
    <li>python38Packages.djangorestframework-jwt</li>
    <li>python38Packages.exchangelib</li>
    <li>python38Packages.flask-appbuilder</li>
    <li>python38Packages.globus-sdk</li>
    <li>python38Packages.twilio</li>
    <li>python39Packages.atlassian-python-api</li>
    <li>python39Packages.djangorestframework-jwt</li>
    <li>python39Packages.flask-appbuilder</li>
    <li>python39Packages.globus-sdk</li>
    <li>python39Packages.twilio</li>
  </ul>
</details>
<details>
  <summary>344 packages skipped due to time constraints:</summary>
  <ul>
    <li>apache-airflow</li>
    <li>appdaemon</li>
    <li>apprise (python38Packages.apprise)</li>
    <li>azure-cli</li>
    <li>beets</li>
    <li>beetsExternalPlugins.alternatives</li>
    <li>beetsExternalPlugins.copyartifacts</li>
    <li>beetsExternalPlugins.extrafiles</li>
    <li>breezy (python38Packages.breezy)</li>
    <li>cabal2nix</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>86 packages built successfully:</summary>
  <ul>
    <li>alerta-server</li>
    <li>buildbot (python38Packages.buildbot)</li>
    <li>buildbot-full (python38Packages.buildbot-full)</li>
    <li>buildbot-ui (python38Packages.buildbot-ui)</li>
    <li>cligh</li>
    <li>cloud-init</li>
    <li>eolie</li>
    <li>mathlibtools (python38Packages.mathlibtools)</li>
    <li>matrix-synapse</li>
    <li>pleroma-bot</li>
    <li>python38Packages.PyGithub</li>
    <li>python38Packages.adal</li>
    <li>python38Packages.azure-core</li>
    <li>python38Packages.azure-datalake-store</li>
    <li>python38Packages.azure-identity</li>
    <li>python38Packages.azure-storage-blob</li>
    <li>python38Packages.criticality-score</li>
    <li>python38Packages.discogs_client</li>
    <li>python38Packages.django-oauth-toolkit</li>
    <li>python38Packages.djangorestframework-simplejwt</li>
    <li>python38Packages.flask-jwt-extended</li>
    <li>python38Packages.gidgethub</li>
    <li>python38Packages.google-auth-oauthlib</li>
    <li>python38Packages.homepluscontrol</li>
    <li>python38Packages.jupyterhub</li>
    <li>python38Packages.keep</li>
    <li>python38Packages.kubernetes</li>
    <li>python38Packages.lazr-restfulclient</li>
    <li>python38Packages.msal</li>
    <li>python38Packages.msal-extensions</li>
    <li>python38Packages.msrest</li>
    <li>python38Packages.msrestazure</li>
    <li>python38Packages.mwoauth</li>
    <li>python38Packages.oauthenticator</li>
    <li>python38Packages.oauthlib</li>
    <li>python38Packages.ondilo</li>
    <li>python38Packages.pyatmo</li>
    <li>python38Packages.pyflume</li>
    <li>python38Packages.pyfxa</li>
    <li>python38Packages.pyjwt</li>
    <li>python38Packages.requests_oauthlib</li>
    <li>python38Packages.ring-doorbell</li>
    <li>python38Packages.snowflake-connector-python</li>
    <li>python38Packages.snowflake-sqlalchemy</li>
    <li>python38Packages.webexteamssdk</li>
    <li>python39Packages.PyGithub</li>
    <li>python39Packages.adal</li>
    <li>python39Packages.azure-core</li>
    <li>python39Packages.azure-datalake-store</li>
    <li>python39Packages.azure-identity</li>
    <li>python39Packages.azure-storage-blob</li>
    <li>python39Packages.buildbot</li>
    <li>python39Packages.buildbot-full</li>
    <li>python39Packages.buildbot-ui</li>
    <li>python39Packages.criticality-score</li>
    <li>python39Packages.discogs_client</li>
    <li>python39Packages.django-oauth-toolkit</li>
    <li>python39Packages.djangorestframework-simplejwt</li>
    <li>python39Packages.exchangelib</li>
    <li>python39Packages.flask-jwt-extended</li>
    <li>python39Packages.gidgethub</li>
    <li>python39Packages.google-auth-oauthlib</li>
    <li>python39Packages.homepluscontrol</li>
    <li>python39Packages.jupyterhub</li>
    <li>python39Packages.keep</li>
    <li>python39Packages.kubernetes</li>
    <li>python39Packages.lazr-restfulclient</li>
    <li>python39Packages.mathlibtools</li>
    <li>python39Packages.msal</li>
    <li>python39Packages.msal-extensions</li>
    <li>python39Packages.msrest</li>
    <li>python39Packages.msrestazure</li>
    <li>python39Packages.mwoauth</li>
    <li>python39Packages.oauthenticator</li>
    <li>python39Packages.oauthlib</li>
    <li>python39Packages.ondilo</li>
    <li>python39Packages.pyatmo</li>
    <li>python39Packages.pyflume</li>
    <li>python39Packages.pyfxa</li>
    <li>python39Packages.pyjwt</li>
    <li>python39Packages.requests_oauthlib</li>
    <li>python39Packages.ring-doorbell</li>
    <li>python39Packages.snowflake-connector-python</li>
    <li>python39Packages.snowflake-sqlalchemy</li>
    <li>python39Packages.webexteamssdk</li>
    <li>sourcehut.dispatchsrht</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `isPy3k`.
Near pkgs/development/python-modules/snowflake-connector-python/default.nix:13:3:
```
   |
13 | , isPy3k
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/snowflake-connector-python/default.nix:73:3:
```
   |
73 |   meta = with lib; {
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/snowflake-connector-python/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/snowflake-connector-python/default.nix:13]: Thanks, I will remove it.

**Comment by dotlambda** [comment]
@fabaff I don't think this needs to go through staging.

**Comment by jonringer** [comment]
~500 python rebuilds is fine. Python packages are relatively "cheap"

**Comment by fabaff** [comment]
Uff, that didn't go as planned.

**Comment by dotlambda** [comment]
This broke NixOps. cc @NixOS/nixops-committers

**Pull request merged**